src/payment/strategies/amazon-pay-v2/amazon-pay-v2-payment-strategy.ts

```diff
@@ -93,7 +93,9 @@ export default class AmazonPayV2PaymentStrategy implements PaymentStrategy {
 
         this._walletButton.click();
 
-        return new Promise<never>(noop);
+        return this._isModalFlow(region)
+            ? Promise.reject(new PaymentMethodCancelledError('Shopper needs to login to Amazonpay to continue'))
```

Comment:
Why do we want reject in this scenario?

If I'm reading it right, we've just opened the modal for the customer and they are probably still in the process of logging in

Reply:
So, rejecting the promise is the way to not trigger the action for the spinner, rejecting will not cause any issue while shopper is in login process. Amazon scripts handles the login, shipping address selection and after that they will hit out backend with all that information, after that moment we will re render UCO with that data. So we are not actually expecting anything at that moment.

Reply:
I thought we were only using the loading spinner if `this._isModalFlow(region)` returned `false`? (line 123)

In which case, this reject is only happening when `this._isModalFlow(region)` returns `true`, which is when the loading spinner is not active?

Reply:
@leeBigCommerce I'm not sure if I understand, yes, we are using the loading spinner in regions that are not US, which is the validation of the function. In this return if the modal flow is true (we are using US region) we need to reject the promise to not trigger the loading spinner.

Reply:
Does AmazonPay provide a callback method when user cancel the flow? As you cancel the flow before user click cancel button. It will be nice if we could hook on the callback method from AmazonPay and implement the related function.

Reply:
Hey @bc-mackxu no, unfortunately they don't. Amazon team is considering adding a callback, but right now they don't offer it.

Reply:
Sorry @deov31 , I've finally grasped that the spinner is triggered outside of this strategy (`checkoutState.statuses.isPending()`) and not by the code in line 123 🤦 

I am a bit concerned that every US execution will result in a `PaymentMethodCancelledError` even when not cancelled, but actually continued.  Though I'm not sure if this will have any actual negative side-effects? (@bc-mackxu, @animesh1987  ?)

We're 'cancelling' not to _cancel_ the payment execution, but to remove a side effect of the pending execution (the spinner).

It's not perfect, but adding a `focus` event listener to the window and tying the promise rejection to that might help minimise these 'false cancelations' - if we need to. The re-focussing of this window doesn't necessarily mean the popup was closed, but it's the closest I think of, given Amazon's API does us no favours.

Reply:
I have similar thoughts as @leeBigCommerce, if we do need to reject the promise then we probably need to create a new error type?

Reply:
@animesh1987 I was wondering the same about @leeBigCommerce's comment, creating new error type seems like a good way to address concerns regarding `PaymentMethodCancelledError`. You guys would be ok with that? Something like `PaymentMethodModalClosed` would be ok?

Reply:
@deov31 what I meant earlier was have you given a thought about what Lee mentioned about adding `focus` listener to the window and dealing with rejections when the focus changes? 

Also `PaymentMethodModalClosed` error doesn't seem right since we are actually opening the modal, if I understand this correctly.

Reply:
hello team @animesh1987 @leeBigCommerce is there any recommendation you could provide us to move forward? 
 or you think should we take this in another direction? from Amazon they suggested us to remove the spinner and we want to provide a solution with the best approach